plugins/hls-explicit-imports-plugin/src/Ide/Plugin/ExplicitImports.hs

```diff
@@ -97,17 +105,35 @@ descriptorForModules recorder modFilter plId =
          -- This plugin provides code lenses
            mkPluginHandler SMethod_TextDocumentCodeLens (lensProvider recorder)
         <> mkResolveHandler SMethod_CodeLensResolve (lensResolveProvider recorder)
-          -- This plugin provides code actions
+        -- This plugin provides inlay hints
+        <> mkPluginHandler SMethod_TextDocumentInlayHint (inlayHintProvider recorder)
+        -- This plugin provides code actions
         <> codeActionHandlers
-
     }
 
+isInlayHintsSupported :: MonadIO m => IdeState -> m Bool
+isInlayHintsSupported state = do
+    Shake.ShakeExtras{lspEnv} <- liftIO $ runAction "" state Shake.getShakeExtras
```

Comment:
This seems a bit complicated, compare with https://github.com/haskell/haskell-language-server/blob/master/ghcide/src/Development/IDE/Plugin/Completions.hs#L208 (I just searched for "clientCaps"!)